### PR TITLE
Add PropTypes to Profile 2.0.

### PR DESCRIPTION
**Pull request by sanlouise**
[TICKET](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9771)

## Description

I added PropTypes to all applicable Profile 2.0 components to make them more maintainable. 

## Testing done
Clicking through the app, I have not run into any PropType errors in the console.

## Acceptance criteria
- [x] Add PropTypes to Profile 2.0 components

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/components/MilitaryInformation.jsx [hunk 1]
```diff
@@ -164,6 +165,18 @@ const MilitaryInformation = ({ militaryInformation }) => (
   </>
 );
 
+MilitaryInformation.propTypes = {
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/components/MilitaryInformation.jsx:168]: @erikphansen When I was adding PropTypes to this component, it appeared that `error` is not a property on `serviceHistory`. Am I missing something or does this mean that there is a bunch of unreachable code in this component?

Comment (erikphansen) [reply src/applications/personalization/profile-2/components/MilitaryInformation.jsx:168]: I believe that gets added only if there is an error fetching the military info data

File: src/applications/personalization/profile-2/components/Profile2Wrapper.jsx [hunk 2]
```diff
@@ -48,4 +49,18 @@ const Profile2 = ({ children, location, routes }) => {
   );
 };
 
+Profile2.propTypes = {
+  children: PropTypes.node.isRequired,
+  location: PropTypes.object,
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/components/Profile2Wrapper.jsx:54]: There is a bug passing `location` currently, once that is fixed, we need to make the PropType required.

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
 
 export function ConnectedAppDeleteModal({
-  appName,
+  title,
   deleting,
   modalOpen,
-  onCloseModal,
-  onConfirmDelete,
+  closeModal,
+  confirmDelete,
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedAppDeleteModal.jsx:11]: you didn't want to keep these in alpha order? :-D

File: src/applications/personalization/profile-2/components/connected-apps/ConnectedApp.jsx [hunk 4]
```diff
@@ -62,10 +62,10 @@ export class ConnectedApp extends Component {
 
           <ConnectedAppDeleteModal
             deleting={this.props.deleting}
-            appName={title}
+            title={title}
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/connected-apps/ConnectedApp.jsx:65]: did you want to keep these in alpha order?

**Review by erikphansen — APPROVED** [review]

**Pull request merged**